README.md

```diff
@@ -1,454 +1,416 @@
+                "type": "dpadFromAxesSource",
+                "xAxisIndex" : 2,
+                "yAxisIndex" : 3
             }
         ]
     }
 }
 ```
 
-#### Touchpad
-A `touchpad` data source maps to two axes in the `Gamepad.axes` array and optionally a button in the `Gamepad.buttons` array.  This mapping is defined by the required `gamepadAxesIndex` in the `xAxis` and `yAxis` properties which represent the indices of the backing entries in the `Gamepad.axes` array.
+### Thumbsticks And Touchpads
+Thumbsticks are a physical part sticking up from the surface of a controller which can be rocked left/right and top/bottom with a circular range.  Often, thumbsticks can also be depressed in a button-like manner.  Touchpads are a physical part, usually circular, that are able to detect the position of a finger placed on their surface.  Often, touchpads can be depressed in the middle in a button-lime manner or at the edges in a dpad-like manner.
 
-The `touchpad.button` must follow the behavior of the `button` data source if it is present.  If `button.supportsTouch` is `true`, the `GamepadButton.touched` property must be `true` when either axis is non-zero.
+Both thumbsticks and touchpads are represented by an entry in the `dataSources` array with the `dataSourceType` property set to `thumbstickSource` and `touchpadSource` respectively.  Both variations must include `xAxisIndex` and `yAxisIndex` properties with values representing the indices in the `Gamepad.axes` array at which to find the related data.  For example:
 
-The following is an example of a `touchpad`:
 ```json
 {
     "gamepad" : {
         "dataSources" : [
             {
-                "touchpad" : {
-                    "id" : "touchpad",
-                    "xAxis" : {
-                        "gamepadAxesIndex" : 2
-                    },
-                    "yAxis" : {
-                        "gamepadAxesIndex" : 3
-                    },
-                    "button" : {
-                        "gamepadButtonsIndex" : 8
-                    }
-                }
+                "id" : "touchpad",
+                "type": "touchpadSource",
+                "xAxisIndex" : 0,
+                "yAxisIndex" : 1
             }
         ]
     }
 }
 ```
 
-### Responses
-Responses are the virtual manifestations of the physical motion controller's changes.  In order to generically reprsent the range of motion particular controllers allow for their various parts, the virtual models contain nodes representing these ranges. There are six currently defined types of virtual responses to physical motion.
+If the thumbstick or touchpad is able to be depressed, the data source must also include a `buttonIndex` property with a value representing the index in the `Gamepad.buttons` array at which to find the button's data. On touchpads with a dpad-like physical behavior, the x-axis and y-axis values can be used to determine which quadrant(s) are being pressed. For example:
```

Comment:
> If the thumbstick or touchpad is able to be depressed...

Maybe "If the thumbstick or touchpad is able to be depressed **or is touch sensitive**"? Even if it's not clickable we'd need the button (with `pressUnsupported: true`) to indicate touched state, right?

Reply:
In the spirit of "no duplicate data", I'd been assuming any non-zero values on the xaxis or yaxis were indicators of a touched state and as a result no button index would be present if it was non-touchable.  Am i missing something?

Reply:
Not necessarily true, since on an Oculus Touch I can rest my thumb on the joystick without moving it, and the deadzone should be generous enough that it would still report `[0, 0]`.

To be clear, I think the schema and logic are right, I'm simply suggesting that this one sentence might need to be expanded. Either way it's a minor difference that doesn't affect any devices I'm aware of (Any that you can think of that have touch-capacitive axis controls that's aren't clickable?)

Reply:
Ah, gotcha.  That's a good point about the deadzone.  And no, I don't know of any like that... I'll fix the sentence and merge.  thanks!!

Reply:
BTW, this isn't just a prose fix.. the schema doesn't have `pressUnsupported` on that type of datasource.  Not a big deal to add it though